src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -98,6 +116,42 @@ impl Future for CloudwatchFuture {
     fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
         loop {
             match &mut self.state {
+                State::DescribeGroup(fut) => {
+                    let response = match ready!(fut.poll_unpin(cx)) {
+                        Ok(response) => response,
+                        Err(error) => {
+                            return Poll::Ready(Err(CloudwatchError::DescribeLogGroups(error)))
+                        }
+                    };
+
+                    if let Some(group) = response
+                        .log_groups
+                        .ok_or(CloudwatchError::NoGroupsFound)?
+                        .into_iter()
+                        .next()
+                    {
+                        debug!(message = "Group found.", group = ?group.log_group_name);
+                        if self.retention_enabled {
+                            if Some(self.retention_days) != group.retention_in_days {
+                                self.state =
+                                    State::PutRetentionPolicy(self.client.put_retention_policy());
+                            }
+                        } else {
+                            self.state =
+                                State::DeleteRetentionPolicy(self.client.delete_retention_policy());
```

Comment:
I think it would be better to not delete the retention policy if one has already been specified. This is the current behaviour and people may be surprised if retention policies start being removed.